src/test/java/Entities/TagTest.java

```diff
@@ -10,4 +10,10 @@ public void testName() {
         TagImpl tag = new TagImpl("name");
         Assertions.assertEquals("name", tag.name());
     }
+    @Test
+    public void testEquals() {
+        TagImpl tag = new TagImpl("name");
+        TagImpl other_tag = new TagImpl("other name");
+        Assertions.assertNotEquals(tag.name(), other_tag.name());
+    }
```

Comment:
This does not test anything. It simply asserts that two names are different. If you meant to test the `equals` method, you would need to utilize that method and show that all reasonable cases where an entity should and shouldn't equal the tag.

Reply:
However, I've made a different PR to add an equals method to the AbstractEntity class instead, so this method should actually be removed from the tag class and from the tests. If we need a tag specific equals we can implement it as needed.

Reply:
I can go ahead and remove this test from the tag class and the tests, but I thought it would be good to keep it in for completion and correctness.

Reply:
Right, but I'm saying that you aren't testing what you say you are testing. You should be doing `assertNotEquals(tag, other_tag);` in order to test the equals method of the tags. Then also add a third tag which has the same name as the first one and make sure those are equal.